README.md
<p >
<a href="https://www.buymeacoffee.com/abhayrawat" target="_blank"><img align="center" src="https://cdn.buymeacoffee.com/buttons/v2/default-yellow.png" alt="Buy Me A Coffee" height="30px" width= "108px"></a>
</p> 

# simple_polls

simple_polls widget is polling widget with language localizations.

Liked my work ? [support me](https://www.buymeacoffee.com/abhayrawat)

## Example
for full example please view example/main.dart
```dart
SimplePollsWidget(
          onSelection: (PollFrameModel model, PollOptions selectedOptionModel) {
            print('Now total polls are : ' + model.totalPolls.toString());
            print('Selected option has label : ' + selectedOptionModel.label);
          },
          model: PollFrameModel(
            title: Container(
              alignment: Alignment.centerLeft,
              child: Text(
                'This is the title of poll. This is the title of poll. This is the title of poll.',
                style: TextStyle(
                  fontSize: 14,
                  fontWeight: FontWeight.w500,
                ),
              ),
            ),
            totalPolls: 100,
            endTime: DateTime.now().toUtc().add(Duration(days: 10)),
            hasVoted: false,
            editablePoll: true,
            options: <PollOptions>[
              PollOptions(
                label: "Option 1",
                pollsCount: 40,
                isSelected: false,
                id: 1,
              ),
              PollOptions(
                label: "Option 2",
                pollsCount: 25,
                isSelected: false,
                id: 2,
              ),
              PollOptions(
                label: "Option 3",
                pollsCount: 35,
                isSelected: false,
                id: 3,
              ),
            ],
          ),
        )
```
## Screenshots

![](https://raw.githubusercontent.com/abhay-s-rawat/simple_poll/main/images/en_options.jpg) ![](https://raw.githubusercontent.com/abhay-s-rawat/simple_poll/main/images/en_results.jpg) ![](https://raw.githubusercontent.com/abhay-s-rawat/simple_poll/main/images/it_options.jpg) ![](https://raw.githubusercontent.com/abhay-s-rawat/simple_poll/main/images/it_results.jpg)

>NOTE:
>Allowed language codes are it,fr,es,gr,en where en is default.
>This widget does not translate title and options, they should be translated by user.

CHANGELOG.md
## 1.0.0

* Initial release.

## 1.0.1

* Minor changes.

## 1.0.2

* Minor changes on readme file, attached screenshots.

## 1.0.3

* Added a functionality to hide undo button on timer expire and added more documentation.

## 1.0.4

* Corrected and added more documentation.

## 1.0.5

* Now onselection method of SimplePollsWidget will return 2 objects as show below. This change was done because a loop was needed to get the selectd option now it just returns the options which registered a tap.
New:
onSelection: (PollFrameModel model, PollOptions selectedOptionModel) {
            print('Now total polls are : ' + model.totalPolls.toString());
            print('Selected option has label : ' + selectedOptionModel.label);
},

Old:
onSelection: (PollFrameModel model) {
        print('Now total polls are : ' + model.totalPolls.toString());
},

## 1.0.6

* Now user can set custom border shape of option and can pass a function onreset which will be called when poll is editable.

example/lib/main.dart
/// This file contains the example showing how to use poll widget in your application.
import 'package:flutter/material.dart';
import 'package:simple_polls/simple_polls.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primaryColor: Colors.indigo),
      debugShowCheckedModeBanner: false,
      home: ExampleApp(),
    );
  }
}

class ExampleApp extends StatelessWidget {
  const ExampleApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(
        title: Text(
          'Polls Widget',
          style: TextStyle(fontSize: 14, color: Colors.white),
        ),
        centerTitle: true,
        toolbarHeight: 40,
      ),
      body: Container(
        child: SimplePollsWidget(
          /// onSelection will be triggered when users presses a option or presses undo button(only available on editable polls).
          /// This function will be called after all the calculation like reducing total polls and marking previous option not selected.
          /// It returns the PollFrameModel .Use this function to do some extra operations like storing this poll into other variable.
          onSelection:
              (PollFrameModel model, PollOptions? selectedOptionModel) {
            print('Now total polls are : ' + model.totalPolls.toString());
            print('Selected option has label : ' + selectedOptionModel!.label);
          },
          onReset: (PollFrameModel model) {
            print(
                'Poll has been reset, this happens only in case of editable polls');
          },
          optionsBorderShape:
              RoundedRectangleBorder(), // Default is stadium border

          /// optionsStyle will have style used in options.
          optionsStyle: TextStyle(
            fontSize: 13,
            fontWeight: FontWeight.w500,
            color: Theme.of(context).primaryColor,
          ),

          /// languageCode will be used to translate some text used in status like total polls, Poll end time, Undo button.
          /// Allowed values are it,fr,es,gr,en where en is default.
          /// For more language support add translations for that language code in translations/translations.dart and widgets/poll_status.dart.
          /// Add timeago.setLocaleMessages('it', timeago.ItMessages()); to register locales for timeago/remaining in widgets/poll_status.dart.
          /// Add 'es': 'deshacer' to maps present in translations/translations.dart for other translations.
          languageCode: 'it',

          /// Content Padding inside polls widget.
          padding: EdgeInsets.all(15),

          /// Margin for polls widget.
          margin: EdgeInsets.all(15),

          /// Data to be passed to polls widget.
          model: PollFrameModel(
            /// Title of the widget.
            title: Container(
              alignment: Alignment.centerLeft,
              child: Text(
                'Questo è il titolo del sondaggio. Questo è il titolo del sondaggio. Questo è il titolo del sondaggio.',
                style: TextStyle(
                  fontSize: 14,
                  fontWeight: FontWeight.w500,
                ),
              ),
            ),
            totalPolls: 100,

            /// Poll end time
            endTime: DateTime.now().toUtc().add(Duration(days: 10)),

            /// If user hasVoted then results screen will show up.
            hasVoted: false,

            /// If poll is editable then undo button will appear in results screen to participate in poll again.
            editablePoll: true,
            options: <PollOptions>[
              /// Configure options here
              PollOptions(
                label: "opzione 1",
                pollsCount: 40,

                /// Polls received by that option.
                isSelected: false,

                /// If poll selected.
                id: 1,

                /// Option id.
              ),
              PollOptions(
                label: "opzione 2",
                pollsCount: 25,
                isSelected: false,
                id: 2,
              ),
              PollOptions(
                label: "opzione 3",
                pollsCount: 35,
                isSelected: false,
                id: 3,
              ),
            ],
          ),
        ),
      ),
    );
  }
}
